assets/javascripts/discourse-assign/initializers/extend-for-assigns.js.es6

```diff
@@ -312,16 +326,70 @@ function initialize(api) {
   api.addKeyboardShortcut("g a", "", { path: "/my/activity/assigned" });
 }
 
+const REGEXP_USERNAME_PREFIX = /^(assigned:)/gi;
+
 export default {
   name: "extend-for-assign",
   initialize(container) {
     const siteSettings = container.lookup("site-settings:main");
     if (!siteSettings.assign_enabled) {
       return;
     }
+    const currentUser = container.lookup("current-user:main");
+    if (currentUser.can_assign) {
+      SearchAdvancedOptions.reopen({
+        _init() {
+          // this.inOptionsForUsers.pushObjects([
+          //   {
+          //     name: I18n.t("search.advanced.in.assigned"),
+          //     value: "assigned"
+          //   },
+          //   {
+          //     name: I18n.t("search.advanced.in.not_assigned"),
+          //     value: "not_assigned"
+          //   }
+          // ]);
+
+          this._super();
+
+          this.set("searchedTerms.assigned", "");
+        },
+
+        @observes("searchedTerms.assigned")
+        updateSearchTermForAssignedUsername() {
+          const match = this.filterBlocks(REGEXP_USERNAME_PREFIX);
+          const userFilter = this.get("searchedTerms.assigned");
+          let searchTerm = this.searchTerm || "";
+          let keyword = "assigned";
+          if (userFilter && userFilter.length !== 0) {
+            if (match.length !== 0) {
+              searchTerm = searchTerm.replace(
+                match[0],
+                `${keyword}:${userFilter}`
+              );
+            } else {
+              searchTerm += ` ${keyword}:${userFilter}`;
+            }
+
+            this.set("searchTerm", searchTerm.trim());
+          } else if (match.length !== 0) {
+            searchTerm = searchTerm.replace(match[0], "");
+            this.set("searchTerm", searchTerm.trim());
+          }
+        },
+
+        _update() {
+          this._super(...arguments);
+          this.setSearchedTermValue(
+            "searchedTerms.assigned",
+            REGEXP_USERNAME_PREFIX
+          );
+        },
+      });
+    }
 
-    withPluginApi("0.8.11", (api) => initialize(api, container));
-    withPluginApi("0.8.28", (api) =>
+    withPluginApi("0.11.0", (api) => initialize(api, container));
+    withPluginApi("0.11.0", (api) =>
```

Comment:
Changing this will stop the plugin working on Discourse Stable. We only need the new API for the advanced search changes in `initialize()`. `registerTopicFooterButtons` can stay on 0.8.28

```suggestion
    withPluginApi("0.8.28", (api) =>
```